pallets/dapp-staking-v3/src/types.rs

```diff
@@ -1902,3 +1839,32 @@ impl CleanupMarker {
             || self.dapp_tiers_index != self.oldest_valid_era
     }
 }
+
+// Wrap the total issuance value with tier percentages.
+pub struct ThresholdsWithIssuance<NT: Get<u32>> {
+    pub thresholds: BoundedVec<TierThreshold, NT>,
+    pub total_issuance: Balance,
+}
+
+impl<NT: Get<u32>> From<ThresholdsWithIssuance<NT>> for BoundedVec<Balance, NT> {
+    fn from(wrapper: ThresholdsWithIssuance<NT>) -> Self {
```

Comment:
I believe this is related to my previous comment [here](https://github.com/AstarNetwork/Astar/pull/1306#discussion_r1695416701)?

If yes, I believe I didn't state my thoughts clearly.

This `ThresholdWithIssuance` struct is very strange, it just seems out of place 🤷.
Previously that was a function that handled this conversion, but now it's done via a separate struct - both approaches take away this _conversion_ from the enum itself.

I'd strongly suggest to remove this struct, and improve the enum with call(s) like this:
```Rust
impl TierThreshold {
    fn threshold(&self, total_issuance: Balance) -> Balance {
        match self {
            TierThreshold::FixedPercentage { .. } => {
                ...
            }
            TierThreshold::DynamicPercentage { .. } => {
                ...
            }
        }
    }
}
``` 

Then you can easily process `TierThresholds` via iter and using `map(...)`.